homeassistant/components/sensor/netatmo.py

```diff
@@ -43,29 +36,9 @@
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
-    """Setup the NetAtmo sensor."""
-    if not validate_config({DOMAIN: config},
-                           {DOMAIN: [CONF_API_KEY,
-                                     CONF_USERNAME,
-                                     CONF_PASSWORD,
-                                     CONF_SECRET_KEY]},
-                           _LOGGER):
-        return None
-
-    import lnetatmo
-
-    authorization = lnetatmo.ClientAuth(config.get(CONF_API_KEY, None),
-                                        config.get(CONF_SECRET_KEY, None),
-                                        config.get(CONF_USERNAME, None),
-                                        config.get(CONF_PASSWORD, None))
-
-    if not authorization:
-        _LOGGER.error(
-            "Connection error "
-            "Please check your settings for NatAtmo API.")
-        return False
-
-    data = NetAtmoData(authorization, config.get(CONF_STATION, None))
+    """Setup the available NetAtmo weather sensors."""
+    netatmo = get_component('netatmo')
+    data = NetAtmoData(netatmo.NETATMO_AUTH, config.get(CONF_STATION, None))
```

Comment:
It actually looks like sensor discovery is triggered automatically. The discovery looks like it will stop here, however, as discovery won't provide config. The name of the station could be provided in the discovery_info to solve this. Maybe at a future time.